lib/slather/project.rb

```diff
@@ -192,10 +257,39 @@ def coverage_service=(service)
         extend(Slather::CoverageService::GutterJsonOutput)
       elsif service == :cobertura_xml
         extend(Slather::CoverageService::CoberturaXmlOutput)
+      elsif service == :html
+        extend(Slather::CoverageService::HtmlOutput)
       else
-        raise ArgumentError, "`#{coverage_service}` is not a valid coverage service. Try `terminal`, `coveralls`, `gutter_json` or `cobertura_xml`"
+        raise ArgumentError, "`#{coverage_service}` is not a valid coverage service. Try `terminal`, `coveralls`, `gutter_json`, `cobertura_xml` or `html`"
       end
       @coverage_service = service
     end
+
+    def configure_binary_file
+      if self.input_format == "profdata"
+        self.binary_file ||= self.class.yml["binary_file"] || File.expand_path(find_binary_file)
+      end
+    end
+
+    def find_binary_file
+      xctest_bundle = Dir["#{profdata_coverage_dir}/**/*.xctest"].reject { |bundle|
+        bundle.include? "-Runner.app/PlugIns/"
+      }.first
+      raise StandardError, "No product binary found in #{profdata_coverage_dir}. Are you sure your project is setup for generating coverage files? Try `slather setup your/project.xcodeproj`" unless xctest_bundle != nil
+
+      # Find the matching binary file
+      xctest_bundle_file_directory = Pathname.new(xctest_bundle).dirname
+      app_bundle = Dir["#{xctest_bundle_file_directory}/*.app"].first
```

Comment:
I found that xctest_bundle_file_directory could have several app bundles. If I run UITest, I will find My-AppUITests-Runner.app and My-App.app